content/articles/creating-an-animated-javafx-field-validation-alert/index.md

```diff
@@ -0,0 +1,758 @@
+
+These are shown below:
+
+```java
+package authentication.login;
+
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.scene.image.Image;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+
+public class HelloApplication extends Application {
+    @Override
+    public void start(Stage stage) throws IOException {
+        FXMLLoader fxmlLoader = new FXMLLoader(HelloApplication.class.getResource("hello-view.fxml"));
+        
+        // Set the scene width and height to 400 and 600 respectively
+        Scene scene = new Scene(fxmlLoader.load(), 400, 600);
+        
+        // Set the name of the window to 'Login Form'
+        stage.setTitle("Login Form");
+        
+        // Add an Icon to the title bar
+        stage.getIcons().add(new Image(HelloApplication.class.getResourceAsStream("icon.png")));
+        stage.setScene(scene);
+        stage.show();
+    }
+
+    public static void main(String[] args) {
+        launch();
+    }
+}
+```
+
+#### Design the view
+
+Open up the `hello-view.fxml` file using the Scene Builder.
+Do the following in it:
+
+- Delete the VBOX and the element in it. Do this by right-clicking on it and selecting delete in the left tab.
+- Drag and drop an AnchorPane where the VBox was deleted. It loads a new AnchorPane that holds other controls in it.
+
+The AnchorPane will have the following:
+
+- **Style**
+- - `-fx-text-fill`:  #8ecae6
+- **Layout**
+- - `Pref Width`: 400
+- - `Pref Height`: 600
+
+The layout makes it fit perfectly with the scene initially created.
+
+- Drag and drop a label from the controls tab into the AnchorPane in the center of the SceneBuilder.
+
+The label will have the following:
+
+- **Properties**
+- - `Text`: LOGIN
+- - `Font`: System 25px Bold
+- **Style**
+- - `-fx-text-fill`:  #219ebc
+- **Layout**
+- - `Pref Width`: 100
+- - `Pref Height`: 20
+- - `Layout X`: 150
+- - `Layout Y`: 40
+
+When viewed using the '**Text**' view, the following code will be found to be automatically generated:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+
+<?import javafx.geometry.*?>
+<?import javafx.scene.control.*?>
+<?import javafx.scene.image.*?>
+<?import javafx.scene.layout.*?>
+<?import javafx.scene.shape.*?>
+<?import javafx.scene.text.*?>
+
+<AnchorPane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="600.0" prefWidth="400.0" style="-fx-background-color: #8ecae6;" xmlns="http://javafx.com/javafx/11.0.2" xmlns:fx="http://javafx.com/fxml/1" fx:controller="authentication.login.HelloController">
+  <!-- Contents inside the XML file -->
+   <children>
+     <!-- Title -->
+      <Label layoutX="150.0" layoutY="40.0" prefHeight="20.0" prefWidth="100.0" style="-fx-text-fill: #219ebc;" text="LOGIN">
+         <font>
+            <Font name="System Bold" size="25.0" />
+         </font></Label>
+   </children>
+</AnchorPane>
+```

```

Comment:
If you already specified the values in the snippet there is no need of writing them separately. Do this fo every instance you have a code snippet.

Reply:
Then that means that its either I will not at all show the code generated or will show a very long code at the end...

Which do you prefer?